### DISCUSSION TITLE
In what context can one use Android ARM64 support?

### DISCUSSION THREAD

**Discussion by martin-honnen** [ikvmnet/ikvm#643] [Q&A]
https://github.com/ikvmnet/ikvm/pull/640 mentions "Android ARM64 support" for "runtime libraries".
What does that mean? 
Can we write MAU Android apps using IKVM? Or AvaloniaUI apps for Android ARM64? 
I am just curious what options that "Android ARM64 support" gives us.

**Comment by wasabii** [comment]
Not really much of anything yet. But the goal would be just to get basic Java code running so they can be deps of .net code.

**Reply by wasabii** [reply]
You're the Saxon guy right? I'd expect that to work. It's just a library. 

I have a blazor branch where I'm working on the same idea there.

**Comment by martin-honnen** [comment]
So I wrote a MAUI project with VS 2022 including the latest IKVM pre and the latest IKVM.Maven to include a MavenReference to Saxon HE 12.5.

Then I tried to debug/run the project, first on the Windows machine where VS runs, that works fine, then on a connected Android device/phone, that fails with

```
System.TypeInitializationException
  message = The type initializer for 'net.sf.saxon.Configuration' threw an exception.
```

Not sure how to debug that further, the Saxon repo code for that class is at https://saxonica.plan.io/projects/saxonmirrorhe/repository/he/revisions/he_mirror_saxon_12_5/entry/src/main/java/net/sf/saxon/Configuration.java.

I also tried to run in the Android emulator but somehow ARM64 Android emulation doesn't work at all, although I even tried on a Windows ARM 64 machine with the ARM64 version of Visual Studio. Tells something about a panic mode that the x64 emulator can't handle ARM64. Not sure how to get it to use a "native" ARM64 emulator given that it is a Windows ARM64 machine.

**Reply by wasabii** [reply]
>Not really much of anything yet.

Doesn't work yet. Lots to work through from the implementation side.

**Reply by martin-honnen** [reply]
Ah, got misled by your comment "You're the Saxon guy right? I'd expect that to work. It's just a library.". 

But then I will wait until there is a release where you announce the support. Or if SaxonCS 13 with a pure .NET HE version comes out first I will see whether that runs cross-platform and on Android, without the need for IKVM.

Thanks for your awesome work on IKVM, anyway.

Seasonal greetings. And a successful 2025.
